### PR TITLE
fix(vitest): logs in `beforeAll` and `afterAll`

### LINKED ISSUE #3366: Logs in beforeAll missing from HTML report
### Describe the bug

When printing logs in the `beforeAll()` block, they are not attributed to the test suite they are contained within. As a result when using the html reporter, log lines are not captured. e.g.:
```
stdout | unknown test
Logging in beforeAll
```
vs
```
stdout | test/suite.test.ts > suite name > foo
Logging in test
```

### Reproduction

Link to [stackbitz](https://stackblitz.com/edit/vitest-dev-vitest-sh82xh?file=test/suite.test.ts)

here you'll see that the logs generated in the beforeall block arent attributed to the suite test, therefore they are not captured in the generated HTML logs

### System Info

```shell
System:
    OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (16) x64 Intel(R) Core(TM) i7-10700K CPU @ 3.80GHz
    Memory: 32.92 GB / 46.84 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    Yarn: 3.2.0 - ~/.yarn/bin/yarn
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
  Browsers:
    Brave Browser: 113.1.51.110
    Chrome: 113.0.5672.63
    Firefox: 112.0.2
  npmPackages:
    @vitest/ui: ^0.31.0 => 0.31.0 
    vitest: ^0.31.0 => 0.31.0
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by fenghan34**
### Description

- Closes #3366 

- I think the reason why logs in hooks `beforeAll` and `afterAll` are missing from the HTML report is because these logs are not attached to the related tasks they were contained in. https://github.com/vitest-dev/vitest/blob/5bb8b38092f107384b0fbfb091d3d60ab4ea39bd/packages/vitest/src/runtime/console.ts#L88

**Here's the comparison between without and with this PR:**

```ts
import { afterAll, beforeAll, describe, expect, test } from 'vitest'

beforeAll(() => {
  console.log('beforeAll')
})

afterAll(() => {
  console.log('afterAll')
})

describe('suite', () => {
  beforeAll(() => {
    console.log('beforeAll')
  })

  afterAll(() => {
    console.log('afterAll')
  })

  describe('nested suite', () => {
    beforeAll(() => {
      console.log('beforeAll')
    })

    afterAll(() => {
      console.log('afterAll')
    })

    test('test', () => {
      expect(true).toBe(true)
    })
  })
})
```

**Before**
image
image

**After**
image
image

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/runtime/runners/test.ts [hunk 1]
```diff
@@ -44,6 +48,8 @@ export class VitestTestRunner implements VitestRunner {
       if (result)
         await rpc().snapshotSaved(result)
     }
+
+    this.workerState.current = suite.suite
```
Comment (sheremet-va) [inline packages/vitest/src/runtime/runners/test.ts:52]: `current` is the test, it shouldn't reference suites. Maybe it's better to add another property for this?

Comment (fenghan34) [reply packages/vitest/src/runtime/runners/test.ts:52]: I noticed that when I was investigating, but I still used it as its type is `Task` here https://github.com/vitest-dev/vitest/blob/9abef3d004a9d3c0d61ae81ecffaa248687ff75d/packages/vitest/src/types/worker.ts#L28 

if `current` should be the current test, how about adding a property `currentSuite`?

**Review by sheremet-va — DISMISSED** [review]

**Comment by sheremet-va** [comment]
@fenghan34 hey! can you fix merge conflicts when you have time, please? 🙏🏻

**Comment by fenghan34** [comment]
> @fenghan34 hey! can you fix merge conflicts when you have time, please? 🙏🏻 

Sure I will do it soon

**Pull request merged**